zip-0312.rst

```diff
@@ -1,7 +1,466 @@
+key will be able to spend the user's funds. For this reason, one interesting
+possibility is to require multiple parties to allow the transaction to go through.
+This can be accomplished with threshold signatures, where the private key is
+split between parties in a way that a threshold (e.g. 2 out of 3) of them must
+sign the transaction in order to create the final signature. This enables scenarios
+such as users and exchanges sharing custody of a wallet, for example.
+
+FROST is one of such threshold signature protocols. However, it can't be used as-is
+since the Zcash protocol also requires re-randomizing public and private keys
+to ensure unlinkability between transactions. This ZIP specifies a variant of
+FROST with re-randomization support.
+
+
+Requirements
+============
+
+- All signatures generated by following this ZIP must be verified successfully
+  as Sapling or Orchard spend authorization signatures using the appropriate
+  validating key.
+- The signatures generated by following this ZIP should meet the security criteria
+  for Signature with Re-Randomizable Keys as specified in the Zcash protocol [#protocol-concretereddsa]_.
+- The threat model described below must be taken into account.
+
+Threat Model
+------------
+
+In normal usage, a Zcash user follows multiple steps in order to generate a
+shielded transaction:
+
+- The transaction is created.
+- The transaction is signed with a re-randomized version of the user's spend
+  authorization private key.
+- The zero-knowledge proof for the transaction is created with the randomizer
+  as an auxiliary (secret) input, among others.
+
+When employing re-randomizable FROST as specified in this ZIP, the goal is to
+split the spend authorization private key :math:`\mathsf{ask}` among multiple
+possible signers. This means that the proof generation will still be generated
+by a single participant, likely the one that created the transaction in the first
+place. Note that this user already controls the privacy of the transaction since
+they are responsible for creating the proof.
+
+This fits well into the "Coordinator" role from the FROST specification [#frost-protocol]_.
+The Coordinator is responsible for sending the message to be signed to all participants,
+and to aggregate the signature shares.
+
+With those considerations in mind, the threat model considered in this ZIP is:
+
+- The Coordinator is trusted with the privacy of the transaction (which includes
+  the unlinkability property). A rogue Coordinator will be able to break
+  unlinkability and privacy, but should not be able to create signed transactions
+  without the approval of `MIN_SIGNERS` participants, as specified in FROST.
+- All key share holders are also trusted with the privacy and of the transaction,
+  thus a rogue key share holder will be able to break its privacy and unlinkability.
+  A future specification may support a scenario where individual key share
+  holders are not trusted with it.
```

Comment:
```suggestion
  holders are not trusted with privacy or unlinkability.
```
Is this possible? Each of the key share holders need to see any transaction they are contributing to signing (less authorizing data, and probably with at least some of the commitments/ciphertexts opened/decrypted so that they can check amounts and recipients), in order to decide whether or not to sign it. Therefore any of them can decide to breach unlinkability by recognizing those transactions on the block chain and publishing that they are linked.

Reply:
I think that if a participant were to publish everything they were sent and their randomness, this would be necessarily be sufficient for a proof (at least, absent changes that we have no idea how to make).